IdentityCore/src/telemetry/request_telemetry/MSIDLastRequestTelemetrySerializedItem.m

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "MSIDCurrentRequestTelemetrySerializedItem+Internal.h"
+#import "MSIDLastRequestTelemetrySerializedItem.h"
+
+@interface MSIDLastRequestTelemetrySerializedItem()
+
+@property (nonatomic) NSArray<MSIDRequestTelemetryErrorInfo *> *errorsInfo;
+
+@end
+
+@implementation MSIDLastRequestTelemetrySerializedItem
+
+- (instancetype)initWithSchemaVersion:(NSNumber *)schemaVersion defaultFields:(NSArray *)defaultFields errorInfo:(NSArray *)errorsInfo platformFields:(NSArray *)platformFields
+{
+    self = [super initWithSchemaVersion:schemaVersion defaultFields:defaultFields platformFields:platformFields];
+    if (self)
+    {
+        _errorsInfo = errorsInfo;
+    }
+    return self;
+}
+
+// Builds last telemetry string using default serialization of each set of fields
+// specified in the last telemetry string schema
+- (NSString *)serialize
+{
+    NSString *telemetryString = [NSString stringWithFormat:@"%@|", self.schemaVersion];
+    telemetryString = [telemetryString stringByAppendingFormat:@"%@|", [super serializeFields: super.defaultFields]];
+    
+    NSUInteger startLength = [telemetryString lengthOfBytesUsingEncoding:NSUTF8StringEncoding];
+    telemetryString = [telemetryString stringByAppendingFormat:@"%@|", [self serializeErrorsInfoWithCurrentStringSize:startLength]];
+    
+    telemetryString = [telemetryString stringByAppendingFormat:@"%@", [super serializeFields: self.platformFields]];
```

Comment:
could platform fields make telemetry string longer than 4k?

Reply:
They definitely could. My thinking is that since we haven't decided what platform fields will be included yet, we don't know how to properly limit them. I thought that platform fields would be limited once we know what they will be.

Reply:
Makes sense. We can revisit later :)